vllm/engine/protocol.py

```diff
@@ -69,24 +71,40 @@ async def beam_search(
         ignore_eos = params.ignore_eos
         temperature = params.temperature
         length_penalty = params.length_penalty
+        include_stop_str_in_output = params.include_stop_str_in_output
 
-        tokenizer = await self.get_tokenizer(lora_request=None)
-        tokenizedPrompt = prompt if isinstance(
-            prompt, list) else tokenizer.encode(prompt)
-        tokenizedLength = len(tokenizedPrompt)
+        tokenizer = await self.get_tokenizer()
+        self.input_preprocessor = InputPreprocessor(model_config,
+                                                    self.tokenizer)
```

Comment:
Btw why is this function defined inside a protocol? Perhaps we should move this to `LLMEngine`? Then we can make use of the existing `input_preprocessor` defined there. @youkaichao

Reply:
Sorry I made a mistake here, I just pushed another commit to change it from `self.input_preprocessor` and `self.tokenizer` to `input_preprocessor` and `tokenizer`.

By the way, in the current release v0.6.3, this function is inside `AsyncLLMEngine` and it has been moved here in a recent pr: [9296](https://github.com/vllm-project/vllm/pull/9296)